src/App.tsx

```diff
@@ -1,60 +1,133 @@
+}));
+
+const usersFromServerFilter = usersFromServer
+  .find(users => todosFromServer
+    .find(todo => users.id === todo.userId)) || null;
 
 export const App = () => {
+  const [name, setName] = useState('');
+  const [userSelect, setUserSelect] = useState(usersFromServerFilter);
+  const [todos, setTodos] = useState<Todo[]>([...preparedTodos]);
+
+  const [formKey, setFormKey] = useState(10);
+  const [isErrorName, setIsErrorName] = useState(false);
+  const [isErrorUser, setIsErrorUser] = useState(false);
+  const [currentUserId, setCurrentUserId] = useState(0);
+
+  const addTodo = (title: string, user: User | null) => {
+    const newTodo = {
+      id: preparedTodos.length + 1,
+      userId: 1,
+      title,
+      user,
+      completed: false,
+    };
+
+    setTodos([...todos, newTodo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setIsErrorName(() => (name === ''));
+
+    setIsErrorUser(() => (currentUserId === 0));
+
+    if (name !== '' && currentUserId !== 0) {
+      addTodo(name, userSelect);
+      setName('');
+      setFormKey(formKey + 1);
+    }
+  };
+
+  const handle = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    event.preventDefault();
+    const { value } = event.target;
+    const needUser = usersFromServer.find(
+      item => item.name === value,
+    );
+
+    setUserSelect(needUser || null);
+    setCurrentUserId(1);
+  };
+
+  const errorTitle = <span className="error">Please enter a title</span>;
+  const errorUser = <span className="error">Please choose a user</span>;
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        key={formKey}
+        action=" /api/users "
+        method=" POST "
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          Name:
+          {' '}
+
+          <input
+            type="text"
+            data-cy="titleInput"
+            placeholder="Enter a name"
+            value={name}
+            onChange={(event) => {
+              setName(event.target.value);
+            }}
+          />
+          {isErrorName && errorTitle}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          User:
+          <select
+            data-cy="userSelect"
+            name="users"
+            id="0"
+            onChange={handle}
+          >
+            <option>Choose a user</option>
+
+            { usersFromServer
+              .map(users => (
+                <>
+                  <option
+                    key={users.id}
+                  >
+                    {users.name}
+                  </option>
+                </>
+              ))}
```

Comment:
we need a react fragment only in the case when we need to return more than one element, if we already have 1 element, then we do not need a react fragment

```suggestion
            {usersFromServer.map(users => (
                  <option
                    key={users.id}
                  >
                    {users.name}
                  </option>
              ))}
```